content/docs/1.2.4/volumes-and-nodes/volume-size.md

```diff
@@ -19,29 +19,74 @@ In this section, you'll have a better understanding of concepts related to volum
 - Since all historical data stored in the snapshots and active data will be calculated into the actual size, the final value can be greater than the nominal size.
 - The actual size will be shown only when the volume is running.
 
-## An example that helps understand volume `Size` and volume `Actual size`:
+## Examples
+
+Through examples, we will explain the changes in volume `size` and/or `actual size` after various operations.
+### An Example That Helps Understand the Volume `Actual Size` after Deleting Data in the Filesystem:
+
+In the following example, we will explain why the volume `actual size` cannot be reduced after deleting data from the filesystem and why it might increase after rewriting the file.
+
+1. Create a 100Gi volume with single replica, then attach and mount it on a node.
+    - For this empty volume, the nominal `size` is 100Gi and the `actual size` is almost 0.
+    - There is some meta info in the volume hence the `actual size` is not exactly 0.
+2. Write 50Gi data in the volume mount point. The `actual size` is increased by 50Gi because of the allocated blocks in the replica for the 2Gi data. Meanwhile, `df` command in the filesystem also shows the 50Gi used space.
+3. Delete the 50Gi data. Then, `df` command shows that the used space of the filesystem is nearly 0, but the `actual size` is unchanged.
+
+    - Users can see the volume `actual size` is not shrunk after deleting the 2Gi data. Longhorn is a block-level storage system. Therefore, The deletion in the filesystem only marks the blocks that belong to the deleted file as unused. Currently, Longhorn does not support TRIM/UNMAP operations, so the `discard` mount option or `fstrim` in the filesystem layer cannot relaim the unused blocks. In consequence, the actual size of Longhorn volumes cannot be shrunk in this case.
+
+
+4. Then, rewrite the 50Gi data, and the `df` command in the filesystem shows 50Gi used space again. Howerver, the `actual size` is increased by 50Gi and becomes 100Gi.
+
+    - After deletion, filesystem may or maynot reuse the recently freed blocks from recent deleted files according to the filesystem design and please refer to [Block allocation strategies of various filesystems](https://www.ogris.de/blkalloc). If the volume nominal `size` is 100Gi, the `actual size` in the end would range from 50Gi to 100Gi since the filesystem may or maynot reuse the freed blocks. On the other hand, if the volume nominal `size` is 60Gi, the `actual size` at the end would range from 50Gi to 60Gi, because the filesystem has to reuse the freed blocks in the 2nd round of writing. Thus, allocating an appropriate nominal `size` for a volume that holds heavy writing tasks according to the IO pattern would make disk space usage more efficient.
+
+### An Example That Helps Understand Volume `Size` and `Actual Size`:
```

Comment:
There are overlapping parts between the 2 examples. I think we can combine them and emphasize the following key points (with more illustrations):
1. Dirty block not releasing issue: Longhorn does not support TRIM/UNMAP operations. Hence deleting files from filesystems won't lead to volume actual size decreasing/shrinking. 
2. Dirty block not reusing issue: Deleting then writing new files would lead to the actual size keeps increasing. Since the filesystems may not reuse the dirty blocks (the recently freed blocks from recent deleted files). Thus, allocating an appropriate nominal `size` for a volume that holds heavy writing tasks according to the IO pattern would make disk space usage more efficient. 
3. Deleting snapshots **may** (not guarantee) eliminate the overlapping part of the used blocks (no matter the blocks are recently released blocks by fs or still hold historical data).